js/classes/zone.js

```diff
@@ -0,0 +1,141 @@
+     * @param {function} loadFunction A function which is going to download all necessary assets and add all necessary entities into the zone.
+     */
+    constructor(minBlock, maxBlock, loadFunction) {
+        this.minBlock = minBlock;
+        this.maxBlock = maxBlock;
+        this.load = () => {
+            // Clear all entities from previous Zone.
+            GAME.clearEntities();
+            // Clear all images/music from previous Zone.
+            ASSET_MGR.clearCache();
+            loadFunction();
+        };
+    };
+
+    // DYNAMIC GETTERS:
+
+    /**
+     * @returns {Vector} The Min (upper left) block that is visible in this Zone.
+     */
+    get MIN_BLOCK() {
+        return this.minBlock;
+    };
+
+    /**
+     * @returns {Vector} The Max (bottom right) block that is visible in this Zone.
+     */
+    get MAX_BLOCK() {
+        return this.maxBlock;
+    };
+
+    /**
+     * @returns {Vector} The minimum (x, y) coordinate visible in this Zone.
+     */
+    get MIN_PT() {
+        return Vector.multiply(this.MIN_BLOCK, Block.SCALED_SIZE);
+    };
+
+    /**
+     * @returns {Vector} The maximum (x, y) coordinate visible in this Zone.
+     */
+    get MAX_PT() {
+        // I want to actually be able to SEE the final block (not just touch its top left corner). 
+        // Therefore:
+        const ones = new Vector(1, 1);
+        return Vector.multiply(Vector.add(this.MAX_BLOCK, ones), Block.SCALED_SIZE);
+    };
+
+    // I am not sure if we are actually going to need the following two getters, but I include them just in case:
+
+    /**
+     * @returns {Vector} The size of the zone, in BLOCKS.
+     */
+    get BLOCK_SIZE() {
+        // We gotta include an extra block to account for 0.
+        const ones = new Vector(1, 1);
+        return Vector.add(new Vector(this.MAX_BLOCK.x - this.MIN_BLOCK.x, this.MAX_BLOCK.y - this.MIN_BLOCK.y), ones);
+    };
+
+    /**
+     * @returns {Vector} The size of the zone, in PIXELS.
+     */
+    get PIXEL_SIZE() {
+        return Vector.multiply(this.BLOCK_SIZE, Block.SCALED_SIZE);
+    };
+
+    /**
+     * This is the only way which we should be externally generating Zones! Navigate this object to set ZONE to the one you want,
+     * and then call its load() function in order to get everything properly loaded for your new Zone.
+     * @returns An object containing ALL ZONES!
+     */
+    static getZones() {
+        // Right now, I am creating zones with no real idea how big they ought to be/what ought be in them.
+        // Therefore, I am defining these constants for MIN/MAX_BLOCK.
+        const zeros = new Vector(0, 0);
+        const defaultMaxBlock = new Vector(100, 25);
+
+
+        /*
+            NOTE: EVERYBODY WILL BE PUTTING CODE HERE! PLEASE:
+
+            (A)     Keep it sorted appropriately. There should only ever be the seven dimensions - playground + the six in-game ones.
```

Comment:
Did we not want to include dimensions for each of us as separate playgrounds as we previously discussed? We don't have to but I am wondering if we want to still. @B1GD3VD0G

Reply:
Oh we're doing zones for that aren't we.

Reply:
Everybody has their own Zones in Playground. Everybody has the ability to make whatever zones they want, as well. Keep the zones which are not intended to be in the game, in the playground "dimension"